hs/src/Reach/Verify/SMT.hs

```diff
@@ -1190,6 +1190,17 @@ smt_e at_dv mdv de = do
       mapM_ (bound at <=< smt_v at) lv
     DLE_setApiDetails {} -> mempty
     DLE_GetUntrackedFunds at _ _ -> unbound at
+    DLE_DataTag at d -> do
+      d' <- smt_a at d
+      n <- smtTypeSort $ argTypeOf d
+      let variants = map fst $ M.toAscList $ dataTypeMap $ argTypeOf d
+      let variantCtors = map (\v -> n <> "_" <> v) variants
+      let variantVs = map (\vCtor -> Atom $ vCtor <> "_pv") variantCtors
+      let variantPs = map (\(vV, vC) -> List [Atom vC, vV])
+           $ zip variantVs variantCtors
+      let variantCs = map (\(vP, i) -> List [vP, Atom $ show i])
+           $ zip variantPs ([0 ..] :: [Int])
+      bound at $ smtApply "match" [d', List variantCs]
```

Comment:
This should probably be a function that gets defined with the data type and then we just call it